### ISSUE TITLE
generic mapped types with <string> resolved as string

### ISSUE THREAD

**Issue by kalleguld** [microsoft/TypeScript#32428]
**TypeScript Version:**  3.5.3

**Search Terms:**
generic mapped type &lt;string&gt; &lt;number&gt; built-in inferred

**Code**

```ts
type SomeType<T> = {
    [K in keyof T]: {foo: boolean}
}

//works with "normal" types
declare var someHtmlType: SomeType<HTMLElement>;

if (someHtmlType && someHtmlType.baseURI.foo){
}

//unexpected error: st is treated as a string, not a SomeType<string>
declare var st: SomeType<string>;

//unexpected compile error: Property 'foo' does not exist on type 'number'
if (st.length.foo){   
}

//unexpected error: st2 is also treated as a string
function f(st2: SomeType<string>){
    
    //unexpected compile error: Property 'foo' does not exist on type '(start: number, end?: number | undefined) => string'.
    if (st2.substring.foo){
        
    }
}

//unexpected: also doesn't work with number
declare var nt: SomeType<number>;
if (nt && nt.toExponential.foo){
}
```

**Expected behavior:**
the variables `st` and `st2` is of type `SomeType<string>`
The variable `nt` is of type `SomeType<number>`
The code example compiles.

**Actual behavior:**
The variables `st` and `st2` is of type `string`
The variable `nt` is of type `number`
I get 3 compile errors.

**Playground Link:** 
[here](http://www.typescriptlang.org/play/#code/C4TwDgpgBAyg9gWwgFXBAPMgfFAvFAbwCgpSoBtAaSgEsA7KAawhDgDMpkBdALkLbhw+AI0EAbCAEM6AXyJyiAekUB3OACdGAZygqawABZQARHQ0JJY41FCQtRACYQAxmMnroAN3dQtiCAASwAhiqJB88EhhGAHIALIAMgCiEkh0wFgA3ERENBwAFH5IQSHRUABk5b7+JaFoAHTCkloQAKoASgCS9QJwAJTECsoArnQQAB6QzsAQDlAQ6uoafFrAtDrAHpIzc81Qkr6b9ADmADRQZmsHkSho6KvqJ1iOLm4eUN7qhxH+0fdHdGOWRyIzGkxcOygzkQYBoEnmi2WUAACktIOpQFAAOS9LFQBxwCA6S7zcY0VZQOAMWzQLF0YYIYQLLG5AqreoSQGGHqCAbyEGKUYTKaQhZLdQrYAAJnW+zEfhsW0hewODxORDYo2mNCpUDYhWlPyidzVgKwfLIUBIZFBwohsyhMLh0DFSNRcHRmJxgjxBKJFzgawm5LWupp2IN7mAfHpjIW5wgdAcAH4YwymV8AD5QUZONj0WZ9PA4U3HLH1a2kPJQA0VKqrKX1LTDYSlnn9YiWuRDQVgkWzPiWBV+rR0LFrNSaXT6IyxjMvVzuLw+dJG26QdBzhbA6v5dJ1i7AerAOBJSZUxPAGiWdt8mRAA)
**Related Issues:** 
[Mapped generic type with &lt;any&gt; parameter inferred as "any"](https://github.com/microsoft/TypeScript/issues/19152)

**Comment by jcalz** [comment]
This is the intended behavior when you try to map over primitive types: see #12447
